test/spec/services/homePagePreferenceSpec.js

```diff
@@ -61,7 +61,7 @@ describe("HomePagePreferenceService", function(){
       };
       HomePagePreferenceService.setHomePagePreference(homePagePref);
       HomePagePreferenceService.notifyInvalidProjectHomePage(homePagePref.project);
-      expect(localStorage.getItem("openshift/home-page-pref/")).not.toBeDefined();
+      expect(localStorage.getItem("openshift/home-page-pref/")).toBeFalsy();
```

Comment:
I'm unclear why this should be necessary since we spyOn localStorage.removeItem to delete the key. It should always be undefined?

https://github.com/openshift/origin-web-console/pull/2736/files#diff-2e146cc351b88485c09870a5cc2f86f9R23

https://github.com/openshift/origin-web-console/blob/4e1c80cf375170e976c77ee8910d0aafacf62e16/app/scripts/services/homePagePreference.js#L38-L44

Reply:
@dtaylor113  thoughts?

@spadgett good Q, I'm only aiming to remove the failing test blocker with this PR.

Reply:
I'm able to reproduce locally, but I worry it means `spyOn` is not working. Strange we don't see this in travis or jenkins. Maybe it's a chrome vs firefox thing.

Reply:
Works locally if I switch the browser to Chrome.
It's a reported bug that spyOn localStorage doesn't work in Firefox.  Figuring out a workaround.

Reply:
`spyOn` wont work, mocking `window.localStorage` or `$window.localStorage` also does not work as `window.localStorage`as a data property as not writable and not configurable. 

Neat 😄

Reply:
OK I am good with the change if spyOn is working on other browsers and it makes the tests pass on Firefox.